apps/managedidentity/managedidentity.go

```diff
@@ -178,8 +209,24 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
+func retry(attempts int, delay time.Duration, c ops.HTTPClient, req *http.Request) (*http.Response, error) {
+	resp, err := c.Do(req)
+	if err == nil && !slices.Contains(retryStatusCodes, resp.StatusCode) {
+		return resp, nil // Success
+	}
+	if attempts-1 < 1 {
+		return resp, nil
+	}
+	time.Sleep(delay)
```

Comment:
MSAL must observe Context cancellation while waiting to retry. See for example the [Azure SDK retry policy](https://github.com/Azure/azure-sdk-for-go/blob/5635b52606e8f156575c11317b87a2d51013b710/sdk/azcore/runtime/policy_retry.go#L212-L219)